lib/generators/rolemodel/modals/templates/app/javascript/packs/rolemodel-modal.js

```diff
@@ -0,0 +1,54 @@
+import makeFormsRemote from './makeFormsRemote'
+
+export default class RolemodelModal {
+  static init() {
+    const rolemodelModal = new RolemodelModal()
+    rolemodelModal.enableModalOpen()
+    rolemodelModal.enableModalClose()
+  }
+
+  get modal() {
+    return document.getElementById('modal')
+  }
+
+  get modalBody() {
+    return document.getElementById('modal-body')
+  }
+
+  get closeModalElement() {
+    return document.getElementById('modal-close')
+  }
+
+  enableModalOpen() {
+    document.querySelectorAll('a[data-custom-remote-modal]').forEach((element) => {
+      element.addEventListener('ajax:send', (e) => {
+        this.modal.classList.add('modal--active')
+      })
+
+      element.addEventListener('ajax:success', (e) => {
+        this.modalBody.innerHTML = e.detail[0].body.innerHTML
+        makeFormsRemote(this.modalBody, 'modal')
+      })
+
+      element.addEventListener('ajax:error', (e) => {
+        this.displayError()
+      })
+    })
+  }
+
+  // Prevent trying to attach an event listener when the page doesn't have the modal.
+  // For example login, etc
+  enableModalClose() {
+    if (this.closeModalElement) {
+      this.closeModalElement.addEventListener('click', (e) => {
+        e.preventDefault()
+        this.modal.classList.remove('modal--active')
+        setTimeout(() => { this.modalBody.innerHTML = ''}, 1000)
+      })
+    }
+  }
+
+  displayError() {
+    this.modalBody.innerHTML = '<div class="page-load__error"><span class="material-icons">error</span><p>Uh oh! Something broke. Try reloading...</p></div>'
```

Comment:
Would it be possible to declare this as a template (JS template literal) or partial somewhere? Just seeing that this is repeated at least once

Reply:
Done